### PR TITLE
upgrade alibi to 0.6.2

### PR DESCRIPTION
**Pull request by sakoush**
**What this PR does / why we need it**:

Upgrade v1 explainer server to use alibi `0.6.2` release

we also move mlserver version to match this upgrade (for v2 protocol)

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
```

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **axsaucedo**
You can assign the PR to them by writing `/assign @axsaucedo` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://jenkins-x.io/v3/develop/reference/chatops/?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/alibi-explain-server/tests/make_test_models.py [hunk 1]
```diff
@@ -27,7 +27,8 @@ def make_anchor_image(dirname: Optional[Path] = None) -> AnchorImage:
 
     # we drop the first batch dimension because AnchorImage expects a single image
     image_shape = model.get_layer(index=0).input_shape[0][1:]
-    alibi_model = AnchorImage(predictor=model, image_shape=image_shape)
+
+    alibi_model = AnchorImage(predictor=model.predict, image_shape=image_shape)
```
Comment (cliveseldon) [inline components/alibi-explain-server/tests/make_test_models.py:31]: Why this change?

Comment (sakoush) [reply components/alibi-explain-server/tests/make_test_models.py:31]: this is because [call](https://www.tensorflow.org/api_docs/python/tf/keras/Model#call) returns a tensor and alibi `0.6.2` will complain that it doesnt return numpy. [predict](https://www.tensorflow.org/api_docs/python/tf/keras/Model#predict) does return numpy.

File: components/alibi-explain-server/pyproject.toml [hunk 2]
```diff
@@ -16,9 +15,8 @@ xgboost = "1.4.2"
 catboost = "0.24.4"
 tornado = "^6.1"
 fastapi = "0.68.1"
-
-shap = { git = "https://github.com/slundberg/shap.git", tag = "v0.39.0" }
 protobuf = "^3.17.3"
+alibi = {extras = ["shap"], version = "^0.6.2"}
```
Comment (RafalSkolasinski) [inline components/alibi-explain-server/pyproject.toml:19]: Let's keep source of dependency as Github (but bump to 0.40.x indeed) as discussed offline

Comment (sakoush) [reply components/alibi-explain-server/pyproject.toml]: Done

**Comment by seldondev** [comment]
@sakoush: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 7805ba1572200212b09daf3604c7aa79f264d636 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3765-integration-2mlxf) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@sakoush: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 7805ba1572200212b09daf3604c7aa79f264d636 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3765-integration-2mlxf) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@sakoush: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | ba7fd44a5bfeb7d489f7c0f8002d7f4ea7642cd5 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3765-integration-w7ld6) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@sakoush: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 812e3c3349e2f5bb4317a2f2b11cfe532f32f674 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3765-integration-r4fdt) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
@sakoush: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 1406e900831f067ff69abe7e3c3e8d2f818cb1f5 | [link](http://dashboard-jx.34.66.17.27.nip.io/namespaces/jx/pipelineruns/seldon-core-pr-3765-integration-76mzm) | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
